File: python/meteroid/models/invoice_documents_event.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .event_id import EventId
from .event_type import EventType
from .invoice_documents_event_data import InvoiceDocumentsEventData


@dataclasses.dataclass
class InvoiceDocumentsEvent(BaseModel):
    _FLATTENED: t.ClassVar[t.Tuple[str, ...]] = ("invoice_documents_event_data",)

    invoice_documents_event_data: InvoiceDocumentsEventData

    id: EventId

    timestamp: datetime

    type: EventType

File: go/invoice_documents_event.go
// this file is @generated
package meteroid

import "time"

type InvoiceDocumentsEvent struct {
	InvoiceDocumentsEventData

	Id EventId `json:"id"`

	// RFC 3339 timestamp.
	Timestamp time.Time `json:"timestamp"`

	Type EventType `json:"type"`
}
